test/type/self.swift

```diff
@@ -299,3 +299,18 @@ struct OuterType {
     return optional.map { `Self`(string: $0) }
   }
 }
+
+// rdar://69804933 - CSApply assigns wrong type to MemberRefExpr for property with
+// DynamicSelfType
+class HasDynamicSelfProperty {
+  var me: Self {
+    return self
+  }
+}
+
+// SILGen doesn't care about the MemberRefExpr's type, so it's hard to come up with an
+// example that actually fails. Here, the rgue 'Self' type was diagnosed as being invalid
```

Comment:
@slavapestov "rgue" here is a typo? Or some acronymous?

Reply:
Typo, should be "rogue". Thanks!